src/xml/builder.cr

```diff
@@ -44,14 +44,14 @@ struct XML::Builder
 
   # Emits the start of an element.
   def start_element(name : String) : Nil
-    element_name = string_to_unsafe name
-    raise XML::Error.new("Invalid element name: '#{name}'", __LINE__) if LibXML.xmlValidateNameValue(element_name).zero?
-    call StartElement, element_name
+    raise XML::Error.new("Invalid element name: '#{name}'", __LINE__) unless validate_element_name name
+    call StartElement, name
   end
 
   # Emits the start of an element with namespace info.
   def start_element(prefix : String?, name : String, namespace_uri : String?) : Nil
-    raise XML::Error.new("Invalid element name: '#{prefix}#{name}'", __LINE__) if LibXML.xmlValidateNameValue(string_to_unsafe("#{prefix}#{name}")).zero?
+    raise XML::Error.new("Invalid element name: '#{name}'", __LINE__) unless validate_element_name name
```

Comment:
Sorry I forgot to mention this but the `line` argument passed to `XML::Error` is the line number in the XML document. I think that's not available when building so I would pass zero here (just like we pass zero in other places). Probably something to improve later on.

Another idea: have a `check_valid_element_name(name, element_kind)` which does the checking and the raising, so for example like `check_valid_element_name(name, "element name")` and `check_valid_element_name(prefix, "prefix")`.